independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyTreeResolver.java

```diff
@@ -281,15 +281,19 @@ private void collectCompileOnly(CollectRequest collectRtDepsRequest, DependencyN
         int flags = DependencyFlags.DIRECT | DependencyFlags.COMPILE_ONLY;
         while (children != null) {
             for (DependencyNode node : children) {
-                if (appBuilder.getDependency(getKey(node.getArtifact())) == null) {
-                    var dep = newDependencyBuilder(node, resolver).setFlags(flags);
-                    if (getExtensionInfoOrNull(node.getArtifact(), node.getRepositories()) != null) {
+                var extInfo = getExtensionInfoOrNull(node.getArtifact(), node.getRepositories());
+                var dep = appBuilder.getDependency(getKey(node.getArtifact()));
```

Comment:
FWIW, this `var` thing makes it a lot harder to review PR as you don't have the IDE context. So you can't easily figure out which class it is and have a look if needed.